File: app/src/main/java/com/shuniuyun/material/adapter/AppBarLayoutPagerAdapter.java
package com.shuniuyun.material.adapter;


import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import com.shuniuyun.material.fragment.AppBarLayoutFragment;
import com.shuniuyun.material.fragment.AppBarLayoutFragment2;

public class AppBarLayoutPagerAdapter extends FragmentStatePagerAdapter {
	private int tabNum;
	public AppBarLayoutPagerAdapter(FragmentManager fm, int tabNum) {
		super(fm);
		this.tabNum=tabNum;
	}

	public Fragment getItem(int position) {
		switch (position) {
		case 0:
			return new AppBarLayoutFragment();
		case 1:
			return new AppBarLayoutFragment2();
		default:
			return null;
		}
	}

	public int getCount() {
		return tabNum;
	}

}

File: app/src/main/java/com/shuniuyun/material/adapter/AppBarLayoutPagerAdapter.kt
package com.shuniuyun.material.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.shuniuyun.material.fragment.AppBarLayoutFragment
import com.shuniuyun.material.fragment.AppBarLayoutFragment2

/**
 *  Created by wangyanbin
 */
class AppBarLayoutPagerAdapter(activity: FragmentActivity) : FragmentStateAdapter(activity) {

    override fun getItemCount() = 2

    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> AppBarLayoutFragment()
            1 -> AppBarLayoutFragment2()
            else -> Fragment()
        }
    }

}